app/queries/district_analytics_query.rb

```diff
@@ -73,6 +91,34 @@ def patients_with_bp_by_period
     group_by_facility_and_date(@patients_with_bp_by_period, :patients_with_bp_by_period)
   end
 
+  def follow_up_patients_by_period
+    #
+    # this is similar to what the group_by_period query already gives us,
+    # however, groupdate does not allow us to use these "groups" in a where clause
+    # hence, we've replicated its grouping behaviour in order to remove the patients
+    # that were registered prior to the period bucket
+    #
+    date_truncate_string =
+      '(DATE_TRUNC("#{@period}", blood_pressures.recorded_at::timestamptz AT TIME ZONE "#{Groupdate.time_zone || 'Etc/UTC'}"))'
+
+    @follow_up_patients_by_period ||=
+      BloodPressure
+        .left_outer_joins(:user)
+        .left_outer_joins(patient: [:medical_history])
+        .joins(:facility)
+        .where(facilities: { id: facilities })
+        .where(deleted_at: nil)
+        .where("medical_histories.hypertension = ?", "yes")
+        .group('facilities.id')
+        .group_by_period(@period, 'blood_pressures.recorded_at')
+        .where("patients.recorded_at < #{date_truncate_string}")
+        .order('facilities.id')
+        .distinct("recorded_at")
```

Comment:
@harimohanraj89 I added `distinct("recorded_at")` so that multiple BPs recorded for a patient in a single day are counted once.